lib/sync-inflate.js

```diff
@@ -17,6 +17,10 @@ function Inflate(opts) {
 
   zlib.Inflate.call(this, opts);
 
+  // Node 8 --> 9 compatibbility fix check
+  this._offset = this._offset === undefined ? this._outOffset : this._offset;
+  this._buffer = this._buffer || this._outBuffer;
+
```

Comment:
Need to update L110 as well.

what writeSync used to return is stored in self._writeState instead.

Reply:
Thank you for this PR btw :)

Reply:
I am assigning the the offset and the buffer to the previous values to make it backward compatible. 
Since the values are the same as in previous versions of Node do we need to change L110?

Reply:
im coudnt figure out where I need to change in `lib/parser-async.js` to support the async case